### ISSUE TITLE
Build failure: linuxPackages_6_6.evdi

### ISSUE THREAD

**Issue by Infinidoge** [NixOS/nixpkgs#265868]
### Steps To Reproduce

1. Rebuild with `"displaylink"` video driver enabled with the latest `nixos-unstable` and `linuxPackages_latest`
2. Observe evdi failing to build

The problem does not occur with `linuxPackages_6_5`

### Build log

The full log is here: https://gist.github.com/Infinidoge/da434e83cb8cf3a77d3a1f0bcce4fb91
Fair warning, it is so long GitHub truncated it. Given it is 7MB, that is expected.

For the sake of making it actually readable, I used an overlay to disable the warnings that filled the log:
```nix
final: prev: {
  linuxPackages_latest = prev.linuxPackages_latest.extend (lfinal: lprev: {
    evdi = lprev.evdi.overrideAttrs (o: {
      env = o.env // { NIX_CFLAGS_COMPILE = o.env.NIX_CFLAGS_COMPILE + " -Wno-unused-parameter -Wno-sign-compare"; };
    });
  });
}
```

Which resulted in this log:
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/cjaxqi2r3573s1mrwc36kh8whmb5y9n9-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j8 SHELL=/nix/store/lf0wpjrj8yx4gsmw2s3xfl58ixmqk8qa-bash-5.2-p15/bin/bash O=\$\(buildRoot\) CC=/nix/store/zlzz2z48s7ry0hkl55xiqp5a73b4mzrg-gcc-wrapper-12.3.0/bin/cc HOSTCC=/nix/store/zlzz2z48s7ry0hkl55xiqp5a73b4mzrg-gcc-wrapper-12.3.0/bin/cc HOSTLD=/nix/store/h8d2j0prdf7pnpgyrgkxrrbfwnmxbq6y-binutils-wrapper-2.40/bin/ld ARCH=x86_64 KVER=6.6.0 KDIR=/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/build
CFLAGS="-isystem./include -isystem./include/uapi -Werror -Wextra -Wall -Wno-error=missing-field-initializers -Werror=sign-compare -Wmissing-prototypes -Wstrict-prototypes -Werror=discarded-qualifiers " make -C module -j8 --jobserver-auth=fifo:/build/GMfifo35
make[1]: Entering directory '/build/source/module'
make[1]: warning: -j8 forced in submake: resetting jobserver mode.
make -C /nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/build M=$PWD
make[2]: Entering directory '/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/build'
  CC [M]  /build/source/module/evdi_platform_drv.o
  CC [M]  /build/source/module/evdi_platform_dev.o
  CC [M]  /build/source/module/evdi_sysfs.o
  CC [M]  /build/source/module/evdi_modeset.o
  CC [M]  /build/source/module/evdi_connector.o
  CC [M]  /build/source/module/evdi_encoder.o
  CC [M]  /build/source/module/evdi_drm_drv.o
  CC [M]  /build/source/module/evdi_fb.o
  CC [M]  /build/source/module/evdi_gem.o
  CC [M]  /build/source/module/evdi_painter.o
[01m[K/build/source/module/evdi_drm_drv.c:127:31:[m[K [01;31m[Kerror: [m[K'[01m[Kdrm_gem_prime_fd_to_handle[m[K' undeclared here (not in a function)
  127 |         .prime_fd_to_handle = [01;31m[Kdrm_gem_prime_fd_to_handle[m[K,
      |                               [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  CC [M]  /build/source/module/evdi_params.o
[01m[K/build/source/module/evdi_fb.c:[m[K In function '[01m[Kevdifb_create[m[K':
[01m[K/build/source/module/evdi_fb.c:421:23:[m[K [01;31m[Kerror: [m[K'[01m[KFBINFO_DEFAULT[m[K' undeclared (first use in this function)
  421 |         info->flags = [01;31m[KFBINFO_DEFAULT[m[K;
      |                       [01;31m[K^~~~~~~~~~~~~~[m[K
[01m[K/build/source/module/evdi_fb.c:421:23:[m[K [01;36m[Knote: [m[Keach undeclared identifier is reported only once for each function it appears in
  CC [M]  /build/source/module/evdi_cursor.o
make[4]: *** [/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/source/scripts/Makefile.build:243: /build/source/module/evdi_fb.o] Error 1
make[4]: *** Waiting for unfinished jobs....
[01m[K/build/source/module/evdi_drm_drv.c:129:31:[m[K [01;31m[Kerror: [m[K'[01m[Kdrm_gem_prime_handle_to_fd[m[K' undeclared here (not in a function)
  129 |         .prime_handle_to_fd = [01;31m[Kdrm_gem_prime_handle_to_fd[m[K,
      |                               [01;31m[K^~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
make[4]: *** [/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/source/scripts/Makefile.build:243: /build/source/module/evdi_drm_drv.o] Error 1
make[3]: *** [/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/source/Makefile:1913: /build/source/module] Error 2
make[2]: *** [/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/source/Makefile:234: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/v1gsjpiyndy6x8hagy4x0ykp7z27y9i1-linux-6.6-dev/lib/modules/6.6.0/build'
make[1]: *** [Makefile:86: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:10: all] Error 2
```
(Sorry for the broken characters, it apparently decided that it wanted to include color despite piping the output...)

### Additional context

Might be a similar issue to #225178, but since it doesn't seem to have a full log attached, it's hard to tell.

### Notify maintainers

There aren't any maintainers of evdi (somewhat concerning) so pinging the latest committers: 
@anthonyroussel @rowanG077
And the former maintainer (sorry):
@eyJhb 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.9, NixOS, 23.11 (Tapir), 23.11.20231104.85f1ba3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
```
The flake lock used is from this commit of my configuration: https://github.com/Infinidoge/universe/commit/47ca9ecc35d58465323a9e6bd205032628f6666d

**Comment by rowanG077** [comment]
This is probably an upstream issue. The linux kernel API changed and evdi doesn't have support yet.

**Comment by anthonyroussel** [comment]
Hello. Thanks for reporting this issue.
Indeed evdi does not support Linux >= v6.5 upstream, so I marked the evdi Kernel module broken with Linux 6.6: https://github.com/NixOS/nixpkgs/pull/266060

**Comment by requaos** [comment]
The upstream fix has been merged to `devel` branch of the project: https://github.com/DisplayLink/evdi/commit/495e8cffbf62f6809108442999084213b014e9e0 which unblocks kernel 6.6 module building. Do we reopen this issue or just open a new issue?

**Comment by Infinidoge** [comment]
I'd say reopen, it helps keep the context together.

**Comment by requaos** [comment]
@alyssais will you kindly reopen this issue? 🙏

**Comment by LDprg** [comment]
When can we expect edvi to work with the 6.6 kernel, since upstream changes are merged? 

I am personally really want to use the 6.6 kernel (mostly because of performance patches and the eevdf scheduler), however since I use displaylink for my monitor (so I can use 180 Hz)  I am currently unable to upgrade.

**Comment by rowanG077** [comment]
While it is fixed upstream it has not yet seen a release. I personally am waiting for the next release.

**Comment by psanford** [comment]
This should be fixed as of #279565

**Comment by anthonyroussel** [comment]
1.14.2 released with official support for Linux => 6.6

* https://github.com/DisplayLink/evdi/releases/tag/v1.14.2
* https://github.com/NixOS/nixpkgs/pull/299660

**Comment by anthonyroussel** [comment]
Was fixed with https://github.com/NixOS/nixpkgs/pull/279565, we can close
